### PR TITLE
[IDSEQ-1075] [Maps] Fix visual styles according to Designs

### PR DESCRIPTION
**Pull request by jshoe**
### Description
- See full list here: https://jira.czi.team/browse/IDSEQ-1075
- Assorted design fixes according to mock here: https://chanzuckerberg.invisionapp.com/d/main#/console/17624454/365364533/preview
- Tighten up map sidebar styles for the sample/project rows
- Adjust styles of the tables, map controls, tooltips, and font sizes.
- Last change is adding RequestContext to PhyloTreeListView (so that it can be properly loaded for its metadata sidebar). Last one!
- Also change it so that in the MapPreviewSidebar, when you select a project, it'll exit the map view and exit preview mode to take you to the whole samples list.

### Demo
Screen Shot 2019-06-18 at 6 28 13 PM
Screen Shot 2019-06-18 at 6 28 20 PM
Screen Shot 2019-06-18 at 6 30 19 PM

### Tests
- Go to the data discovery views, go to the map, select locations, observe the updated visual styles above.

### CODE REVIEW

File: app/assets/src/components/views/discovery/BaseDiscoveryView.jsx [hunk 1]
```diff
@@ -16,17 +17,19 @@ class BaseDiscoveryView extends React.Component {
...
+      rowClassName,
+      rowHeight,
     } = this.props;
 
     return (
       <Table
         columns={columns}
         data={data}
-        defaultRowHeight={68}
+        defaultRowHeight={rowHeight || 68}
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/BaseDiscoveryView.jsx]: Do you know the origin of the 68?
Can you switch it to a prop with default value ?

File: app/assets/src/components/views/discovery/DiscoveryView.jsx [hunk 2]
```diff
@@ -608,13 +608,15 @@ class DiscoveryView extends React.Component {
     const { mapSidebarTab } = this.state;
     this.setState(
       {
+        currentDisplay: "table",
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/DiscoveryView.jsx]: Can we select projects from the map?

File: app/assets/src/components/views/discovery/mapping/base_map.scss [hunk 3]
```diff
@@ -10,8 +10,12 @@
     position: absolute;
     bottom: 50px;
     right: 15px;
+    box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.1);
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/mapping/base_map.scss]: If we can't reuse a box-shadow already defined in `_elements` (always double check with Jenn to push for consistency), move this definition there and use a variable here.

File: app/assets/src/components/views/discovery/mapping/base_map.scss [hunk 4]
```diff
@@ -10,8 +10,12 @@
...
 
     button {
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/mapping/base_map.scss]: Should this use a class instead? Or is part of 3rd party lib?

File: app/assets/src/components/views/discovery/mapping/map_preview_sidebar.scss [hunk 5]
```diff
@@ -48,7 +48,24 @@
 
 .tableDataRow {
   align-items: flex-start;
-  padding: 11px 0;
+  padding: 10px 0;
+  margin-left: 0;
+
+  // Override defaults
+  &:global(.ReactVirtualized),
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/discovery/mapping/map_preview_sidebar.scss]: There are props to pass class names to customize this I believe. Can you search within `BaseTable` and `ReactVirtualize` libraries?

File: app/views/phylo_trees/index.html.erb [hunk 6]
```diff
@@ -5,6 +5,7 @@
       taxon: JSON.parse('<%= raw escape_json(@taxon) %>'),
       phyloTrees: JSON.parse('<%= raw escape_json(@phylo_trees) %>'),
       allowedFeatures: JSON.parse('<%= raw escape_json(@current_user.allowed_feature_list) %>'),
-    }, 'PhyloTreeListView_page');
+    }, 'PhyloTreeListView_page',
+    JSON.parse('<%= raw escape_json(request_context)%>'));
```
Comment (tfrcarvalho) [inline app/views/phylo_trees/index.html.erb]: Why did you need to add this on this PR?

File: app/assets/src/components/views/discovery/BaseDiscoveryView.jsx (hunk 1 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/BaseDiscoveryView.jsx]: no idea where the 68 was from actually. Moved to default props 👍

File: app/assets/src/components/views/discovery/DiscoveryView.jsx (hunk 2 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: yes this is referring to clicking on the project in the right pane table while in the map

File: app/assets/src/components/views/discovery/mapping/base_map.scss (hunk 3 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/mapping/base_map.scss]: changed 👍

File: app/assets/src/components/views/discovery/mapping/base_map.scss (hunk 4 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/mapping/base_map.scss]: yes and yes (assuming you meant the button) 👍

File: app/assets/src/components/views/discovery/mapping/map_preview_sidebar.scss (hunk 5 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/mapping/map_preview_sidebar.scss]: yes it looks like we get headerClassName and rowClassName https://github.com/bvaughn/react-virtualized/blob/master/docs/customizingStyles.md

I moved out the header but wasn't able to override rowColumn/headerColumn specifically

Screen Shot 2019-06-19 at 2 46 48 PM

File: app/views/phylo_trees/index.html.erb (hunk 6 above)
Comment (jshoe) [reply app/views/phylo_trees/index.html.erb]: wasn't necessary 👍 probably could've bundled with a different PR

File: app/assets/src/components/views/discovery/DiscoveryView.jsx (hunk 2 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: Shouldnt we continue on the map?

File: app/views/phylo_trees/index.html.erb (hunk 6 above)
Comment (tfrcarvalho) [reply app/views/phylo_trees/index.html.erb]: OK, fine to change if you want to bundle it...

File: app/assets/src/components/views/discovery/DiscoveryView.jsx (hunk 2 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: Jenn determined that it should set to the table view instead https://jira.czi.team/browse/IDSEQ-1087

Comment (jshoe) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: cc @happyimadesignr

Comment (tfrcarvalho) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: 👌
